### PR TITLE
Enforce windows password complexity requirements in acs-engine client…

### PR DESCRIPTION
**Pull request by vpatelsj**
…. #2407

Added Windows Agent password complexity check as per guidelines in  https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/password-must-meet-complexity-requirements.

This will prevent generation of arm templates if password complexity for windows vm does not meet the complexity requirements.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
@vpatelsj Thanks for this! Could you add a unit test for password that doesn't meet the new requirements to ensure it's failing as expected?

**Comment by jackfrancis** [comment]
@PatrickLang FYI

**Comment by vpatelsj** [comment]
@jackfrancis thanks for the review. I will add the negative test case.
@tariq1890 Yes, you are right in pointing that out. I don't have enough domain background on diacritic and cyrillic characters so wanted to leave it out for now. I had pondered adding this expression based on this discussion - https://stackoverflow.com/questions/150033/regular-expression-to-match-non-english-characters

**Comment by tariq1890** [comment]
I'd leave that to you if you'd want to stick with the current implementation or validate the diacritic chars.

If @jackfrancis is okay with going forward with the current implementation, we can get this merged in. But I would ask that we provide an error message that is consistent with our implementation and not link to the password complexity docs.

**Comment by vpatelsj** [comment]
@tariq1890 Thanks, you have a valid point about the specifics of password complexity in error message. I think there is value in going ahead with current implementation as it will still with with diacritic and cyrillic passwords as long as their is at least one english char in their password. I would love to comeback and put additional checks in future to match closely with windows once I figure the right way to address those chars.

**Comment by vpatelsj** [comment]
@jackfrancis @tariq1890 I've tried to address your review comments in the new commit. Thanks!

**Comment by CecileRobertMichon** [comment]
Added one comment, otherwise lgtm. Thanks @vpatelsj!

File: pkg/api/vlabs/validate_test.go [hunk 1]
```diff
@@ -1265,6 +1265,20 @@ func TestWindowsVersions(t *testing.T) {
 				"should not error on valid Windows version: %v", err,
 			)
 		}
+		p = getK8sDefaultProperties(true)
```
[the remark this answers is not in the document]
Comment (CecileRobertMichon) [reply pkg/api/vlabs/validate_test.go]: Actually I see that's already covered with the general valid Windows profile. However the length = 0 and name == password cases are both missing.

**Comment by vpatelsj** [comment]
@CecileRobertMichon - added your recommended test cases as well. thanks!

**Comment by tariq1890** [comment]
@vpatelsj Thanks!

**Pull request merged**